web/client/epics/geostory.js

```diff
@@ -461,6 +475,52 @@ export const handlePendingGeoStoryChanges = action$ =>
                     )
             )
     );
+
+/**
+ * Handle the url updates on currentPage change
+ * @param {Observable} action$ stream of actions
+ * @param {object} store redux store
+ */
+export const urlUpdateOnScroll = (action$, {getState}) =>
+    action$.ofType(UPDATE_CURRENT_PAGE)
+        .debounceTime(50) // little delay if too many UPDATE_CURRENT_PAGE actions come
+        .switchMap(({sectionId, columnId}) => {
+            if (
+                modeSelector(getState()) !== 'edit' && (!!columnId || !!sectionId)
+                && updateUrlOnScrollSelector(getState())
+            ) {
+                const urlObject = queryString.parse(window.location.href);
+                const EMPTY = 'EMPTY';
+                if (sectionId) {
+                    if (urlObject.sectionId || urlObject.columnId) {
+                        history.pushState(null, '', replace(window.location.href, /[^\&]+$/, `sectionId=${sectionId}`));
+                    } else {
+                        history.pushState(null, '', `${window.location.href}&sectionId=${sectionId}`);
+                    }
+                } else if (!sectionId && columnId && isString(columnId) && columnId !== EMPTY) {
+                    if (urlObject.sectionId || urlObject.columnId) {
+                        history.pushState(null, '', replace(window.location.href, /[^\&]+$/, `columnId=${columnId}`));
+                    }
+            } else return Observable.empty();
+        }
+            return Observable.empty();
+        });
+
+/**
+ * Loads geostory on POP history
+ * @param {Observable} action$ stream of actions
+ */
+export const test = (action$) =>
```

Comment:
This epic is needed, but it has a bug so i fixed it and share should work fine now. I also renamed the name of epic(because i forgot to do that before)